src/Microsoft.Health.Dicom.Core/Features/Store/IIndexDataStore.cs

```diff
@@ -77,12 +78,14 @@ public interface IIndexDataStore
     /// <param name="partitionKey">The partition key.</param>
     /// <param name="dicomDataset">The DICOM dataset whose status should be updated.</param>
     /// <param name="watermark">The DICOM instance watermark.</param>
+    /// <param name="instanceKey">The instance key.</param>
     /// <param name="queryTags">Queryable dicom tags</param>
+    /// <param name="fileProperties">file properties</param>
     /// <param name="allowExpiredTags">Optionally allow an out-of-date snapshot of <paramref name="queryTags"/>.</param>
     /// <param name="hasFrameMetadata">Has additional frame range metadata stores.</param>
     /// <param name="cancellationToken">The cancellation token.</param>
     /// <returns>A task that represents the asynchronous update operation.</returns>
-    Task EndCreateInstanceIndexAsync(int partitionKey, DicomDataset dicomDataset, long watermark, IEnumerable<QueryTag> queryTags, bool allowExpiredTags = false, bool hasFrameMetadata = false, CancellationToken cancellationToken = default);
+    Task EndCreateInstanceIndexAsync(int partitionKey, DicomDataset dicomDataset, long watermark, long? instanceKey, IEnumerable<QueryTag> queryTags, FileProperties fileProperties = null, bool allowExpiredTags = false, bool hasFrameMetadata = false, CancellationToken cancellationToken = default);
```

Comment:
Perhaps just pass an `InstanceStorageKey`? That way there's also a symmetry to the methods